frontend/src/metabase/admin/performance/utils.ts

```diff
@@ -0,0 +1,129 @@
+import { Cron, weekdays } from "metabase/components/Schedule/constants";
+import type {
+  ScheduleDayType,
+  ScheduleFrameType,
+  ScheduleSettings,
+  ScheduleType,
+} from "metabase-types/api";
+
+const dayToCron = (day: ScheduleSettings["schedule_day"]) => {
+  const index = weekdays.findIndex(o => o.value === day);
+  if (index === -1) {
+    throw new Error(`Invalid day: ${day}`);
+  }
+  return index + 1;
+};
+
+const frameToCron = (frame: ScheduleFrameType) =>
+  ({ first: "1", last: "L", mid: "15" }[frame]);
+
+const frameFromCron = (frameInCronFormat: string) => {
+  const map: Record<string, ScheduleFrameType> = {
+    "15": "mid",
+    "1": "first",
+    L: "last",
+  };
+  return map[frameInCronFormat];
+};
+
+export const scheduleSettingsToCron = (settings: ScheduleSettings): string => {
+  const minute = settings.schedule_minute?.toString() ?? Cron.AllValues;
+  const hour = settings.schedule_hour?.toString() ?? Cron.AllValues;
+  let weekday = settings.schedule_day
+    ? dayToCron(settings.schedule_day).toString()
+    : Cron.NoSpecificValue;
+  const month = Cron.AllValues;
+  let dayOfMonth: string = settings.schedule_day
+    ? Cron.NoSpecificValue
+    : Cron.AllValues;
+  if (settings.schedule_type === "monthly" && settings.schedule_frame) {
+    if (settings.schedule_day) {
+      const frameInCronFormat = frameToCron(settings.schedule_frame).replace(
+        /^1$/,
+        "#1",
+      );
+      const dayInCronFormat = dayToCron(settings.schedule_day);
+      weekday = `${dayInCronFormat}${frameInCronFormat}`;
+    } else {
+      dayOfMonth = frameToCron(settings.schedule_frame);
+    }
+  }
+  const second = "0";
```

Comment:
worlds smallest nit... all your variables are declared in ascending order until you get to second :P